src/Core/Managed/Operation.CC.Shared/TelemetryClientExtensions.cs

```diff
@@ -21,47 +21,95 @@ public static class TelemetryClientExtensions
         /// <returns>Operation item object with a new telemetry item having current start time and timestamp.</returns>
         public static IOperationHolder<T> StartOperation<T>(this TelemetryClient telemetryClient, string operationName) where T : OperationTelemetry, new()
         {
+            return StartOperation<T>(telemetryClient, operationName, operationId: null, parentOperationId: null);
+        }
+
+        /// <summary>
+        /// Start operation creates an operation object with a respective telemetry item. 
+        /// </summary>
+        /// <typeparam name="T">Type of the telemetry item.</typeparam>
+        /// <param name="telemetryClient">Telemetry client object.</param>
+        /// <param name="operationName">Name of the operation that customer is planning to propagate.</param>
+        /// <param name="operationId">Operation ID to set in the new operation.</param>
+        /// <param name="parentOperationId">Optional parent operation ID to set in the new operation.</param>
+        /// <returns>Operation item object with a new telemetry item having current start time and timestamp.</returns>
+        public static IOperationHolder<T> StartOperation<T>(this TelemetryClient telemetryClient, string operationName, string operationId, string parentOperationId = null) where T : OperationTelemetry, new()
```

Comment:
Presumably, if an app wants to flow ```Context.User.Id``` from parent to child, then it would need to install an initializer like ```UserTelemetryInitializer``` from the ASP.NET repo. That seems reasonable to me.